lib/resources/etc_libuser.rb

```diff
@@ -0,0 +1,108 @@
+require 'utils/simpleconfig'
+require 'utils/object_traversal'
+
+module Inspec::Resources
+  class EtcLibUser < Inspec.resource(1)
+    name 'etc_libuser'
+    desc 'Use the etc_libuser InSpec audit resource to check the configuration of /etc/libuser.conf.
+          Due to limitations of the RSpec engine and for consistency, please use the method described here
+          (https://github.com/chef/inspec/issues/1543) for accessing the hierarchical values in
+          /etc/libuser.conf'
+    example "
+          ---start sample /etc/libuser.conf---
+          # See libuser.conf(5) for more information.
+          ...
+          [import]
+          # Data from these files is used when libuser.conf does not define a value.
+          # The mapping is documented in the man page.
+          login_defs = /etc/login.defs
+          default_useradd = /etc/default/useradd
+          ...
+          ---end sample /etc/libuser.conf---
+
+          describe etc_libuser do
+            its(['import','login_defs']){should eq ['/etc/login.defs']}
+          end
+         "
+
+    include ObjectTraverser
+    attr_reader :params, :raw_content
+
+    class UnknownEtcLibUserConf < StandardError; end
+
+    def initialize(conf_path = nil)
+      @conf_path = conf_path || inspec.os_env('LIBUSER_CONF').content || '/etc/libuser.conf'
+      file = inspec.file(@conf_path)
+      return skip_resource "The `etc_libuser` resource is not supported on your OS. Can't find '#{@conf_path}'." unless file.file?
+      @params = nil
+      @raw_content = load_raw_content(@conf_path)
+      @params = parse(@raw_content)
+    end
+
+    def parse(content)
+      @params = SimpleConfig.new(content, multiple_values: true).params
+    end
+
+    def to_s
+      'Libuser Configuration'
+    end
+
+    ###### NOTE: The following methods have been directly copied from the JSON resource due to bug #1300 (https://github.com/chef/inspec/issues/1300) #####
+    
+    # Shorthand to retrieve a parameter name via `#its`.
+    # Example: describe json('file') { its('paramX') { should eq 'Y' } }
+    #
+    # @param [String] name name of the field to retrieve
+
+    # @return [Object] the value stored at this position
+    def method_missing(*keys)
+      # catch bahavior of rspec its implementation
+      # @see https://github.com/rspec/rspec-its/blob/master/lib/rspec/its.rb#L110
+      keys.shift if keys.is_a?(Array) && keys[0] == :[]
+      value(keys)
+    end
+
+    def value(key)
+      # uses ObjectTraverser.extract_value to walk the hash looking for the key,
+      # which may be an Array of keys for a nested Hash.
+      extract_value(key, params)
+    end
+
+    def load_raw_content(opts)
+      # if the opts isn't a hash, we assume it's a path to a file
+      unless opts.is_a?(Hash)
```

Comment:
I'm unclear why this is needed. The resource is written such that only a filename parameter is expected. When would a user supply a Hash? Is this method just copy/pasta from another resource implementation? Looks like it may have been copied from the `json` resource and should probably get removed.

Reply:
Yeah, I directly copied this code from the JSON resource too. Would you like me to remove it now? Or, should I just wait until you determine the best approach for moving forward (e.g. updating SimpleConfig and subclassing INI). I am happy to do whatever you think is best. Thanks!

Reply:
I believe subclassing the `ini` resource and modifying `SimpleConfig` is the right approach. So much of this resource code, including this method, will likely need to be removed. :)